crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -82,19 +65,60 @@ pub(crate) async fn farm(
     })
     .await??;
 
-    let subspace_codec = SubspaceCodec::new(identity.public_key());
+    let (plots, mut farming_plotting): (Arc<Vec<_>>, FuturesUnordered<_>) = {
+        let plot_size = plot_size / PIECE_SIZE as u64;
+        let single_plot_sizes =
+            std::iter::repeat(max_plot_size).take((plot_size / max_plot_size) as usize);
+        let single_plot_sizes = if plot_size % max_plot_size > 0 {
+            single_plot_sizes
+                .chain(std::iter::once(plot_size % max_plot_size))
+                .collect::<Vec<_>>()
+        } else {
+            single_plot_sizes.collect()
+        };
+
+        let (mut plots, farming_plotting) = <(Vec<_>, FuturesUnordered<_>)>::default();
+
+        for (base_directory, max_plot_pieces) in (0..)
+            .map(|i| base_directory.join(format!("plot{i}")))
+            .zip(single_plot_sizes)
+        {
+            let (plot, plotting, farming) = farm_single_plot(
+                base_directory,
+                reward_address,
+                client.clone(),
+                object_mappings.clone(),
+                max_plot_pieces,
+                best_block_number_check_interval,
+            )
+            .await?;
+            plots.push(plot);
+            farming_plotting.push(Box::pin(async move {
+                tokio::select! {
+                    res = plotting.wait() => if let Err(error) = res {
+                        return Err(anyhow!(error))
+                    },
+                    res = farming.wait() => if let Err(error) = res {
+                        return Err(anyhow!(error))
+                    },
+                }
+                Ok(())
+            }));
+        }
+
+        (Arc::new(plots), farming_plotting)
+    };
 
     // Start RPC server
     let ws_server = WsServerBuilder::default()
         .build(ws_server_listen_addr)
         .await?;
     let ws_server_addr = ws_server.local_addr()?;
     let rpc_server = RpcServerImpl::new(
-        farmer_metadata.record_size,
-        farmer_metadata.recorded_history_segment_size,
-        plot.clone(),
+        record_size,
+        recorded_history_segment_size,
+        Vec::clone(&*plots),
```

Comment:
Why not cloning the `Arc` itself?

Reply:
Changed, but i don't think it matters too much. It is either double indirection or cloning multiple `Arc`s.